sklearn/cluster/tests/test_hierarchical.py

```diff
@@ -598,3 +599,16 @@ def increment(self, *args, **kwargs):
     linkage_tree(X, connectivity=connectivity, affinity=fa.increment)
 
     assert_equal(fa.counter, 3)
+
+
+def test_n_components_deprecation():
+    # Test that a Deprecation warning is thrown when n_components_
+    # attribute is accessed
+
+    X = np.array([[1, 2], [1, 4], [1, 0], [4, 2]])
+    agc = AgglomerativeClustering().fit(X)
+
+    with pytest.warns(DeprecationWarning, match='parameter changed'):
+        warnings.warn("No warning", DeprecationWarning)
```

Comment:
please remove this line: you're explicitly raising a warning here. We need to make sure that what raises the warning is the line below (accessing to `n_components_`), not this line.

You can also remove `import warnings`

Also maybe use a longer match so that someone reading the test knows that people should use `n_connected_components_` instead.

Reply:
Got it, should all be fixed in commit [#822eaa5](https://github.com/scikit-learn/scikit-learn/pull/13427/commits/822eaa5240872e71a5b6b898f0db499e6b0a2ac8)